src/main/kotlin/com/openlattice/BackgroundExternalDatabaseUpdatingService.kt

```diff
@@ -0,0 +1,104 @@
+package com.openlattice
+
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.core.IMap
+import com.openlattice.assembler.PostgresDatabases
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.indexing.MAX_DURATION_MILLIS
+import com.openlattice.organization.OrganizationExternalDatabaseColumn
+import com.openlattice.organization.OrganizationExternalDatabaseTable
+import com.openlattice.organizations.ExternalDatabaseManagementService
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import java.util.*
+
+/**
+ *
+ */
+class BackgroundExternalDatabaseUpdatingService(
+        private val hazelcastInstance: HazelcastInstance,
+        private val edms: ExternalDatabaseManagementService
+) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(BackgroundExternalDatabaseUpdatingService::class.java)
+    }
+
+    private val organizationExternalDatabaseColumns: IMap<UUID, OrganizationExternalDatabaseColumn> = hazelcastInstance.getMap(HazelcastMap.ORGANIZATION_EXTERNAL_DATABASE_COlUMN.name)
+    private val organizationExternalDatabaseTables: IMap<UUID, OrganizationExternalDatabaseTable> = hazelcastInstance.getMap(HazelcastMap.ORGANIZATION_EXTERNAL_DATABASE_TABLE.name)
+    private val aclKeys: IMap<String, UUID> = hazelcastInstance.getMap(HazelcastMap.ACL_KEYS.name)
+
+    @Suppress("UNUSED")
+    @Scheduled(fixedRate = MAX_DURATION_MILLIS)
+    fun scanOrganizationDatabases() {
+        val orgIds = edms.getOrganizationIds()
```

Comment:
I would rather use a mapstore here for organization ids (I commented some of them in conductor-client PR) and lock them (see: `BackgroundIndexingService`) or do the query here and put the ids in a queue (see `BackgroundLinkingIndexingService`)

Reply:
👍